components/eamxx/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -386,7 +387,7 @@ void RRTMGPRadiation::run_impl (const double dt) {
   auto d_qv = get_field_in("qv").get_view<const Real**>();
   auto d_qc = get_field_in("qc").get_view<const Real**>();
   auto d_qi = get_field_in("qi").get_view<const Real**>();
-  auto d_cldfrac_tot = get_field_in("cldfrac_tot").get_view<const Real**>();
+  auto d_cldfrac_tot = get_field_in("cldfrac_tot_for_output").get_view<const Real**>();
```

Comment:
Question for @brhillman here: does radiation use total cldfrac for anything other than writing output? Aaron is changing the definition of cloud fraction here...

Reply:
right.  This was my main concern.  I do see that cloud fraction is passed to `rrtmgp::main` as well as used for `cldlow`, `cldmed` and `cldhigh`.  It would not be difficult to use both definitions of cloud frac total if that is warranted.

Reply:
Okay, I misunderstood a little. Cloud fraction is still all or nothing in the presence of ice (0 or 1), this just changes how much ice needs to be present to consider a layer ice filled vs absent of ice. I’m tempted to say we should stick with the same definition as micro and any other processes use for determining whether or not a layer contains ice. But what would be really nice is to have some estimate of how much of the layer actually contains ice, because this will affect the heating. Note that sticking with the mask p3 uses means that cldlow, cldmid, and cldhgh will be trivially 1 in the case of any ice, but using the new “for_output” fields will mean they are sometimes 1, but also sometimes not. I guess that’s the point, but it does change the behavior of radiative heating (i.e., sometimes ice will have no impact on rad now).

Reply:
Sorry for tying you in knots, Ben. I was trying to ask the simpler question of "does radiation use cloud fraction for any of its internal calculations". Since it does, we should definitely pass it the old version of cloud fraction in addition to the "for output only" cloud fraction to use for diagnostics. 

Put another way, the goal of this PR is to fix some bad model outputs so if it is answer changing we've screwed up.

Reply:
Looking at it a bit more, I think we should keep the original threshold for considering a layer to be cloudy in terms of what rad sees. Arbitrarily lowering this threshold would just mean rad would completely ignore more layers that actually contain some amount of ice. With that though, this will keep our cldlow, cldmid, cldhigh definitions the same. We constructed those to be exactly consistent with what radiation sees by vertically projecting the subcolumn cloud mask used to get the subcolumn optical depth profiles that are used for computing rad fluxes. When we did this originally, we hailed this as a great improvement in consistency over how this used to be handled in other models, that used separate params for cloud area. I understand that keeping the current threshold means that we will continue to have a lot of, in particular, cldhigh columns that are entirely cloudy, but since this is what radiation actually sees, I might argue this isn’t such a bad thing. It seems to me that arbitrarily changing this threshold would just “mask” the real issue here, which is that we do not have an estimate of partial cloud ice fraction. At high enough resolution, this probably ceases to be a problem, but at coarser resolutions it is probably a big problem.

Reply:
> Sorry for tying you in knots, Ben. I was trying to ask the simpler question of "does radiation use cloud fraction for any of its internal calculations". Since it does, we should definitely pass it the old version of cloud fraction in addition to the "for output only" cloud fraction to use for diagnostics.
> 
> Put another way, the goal of this PR is to fix some bad model outputs so if it is answer changing we've screwed up.

No worries, I think it’s good to think through these things until we completely understand the choices we have made and whether or not they (still) make sense.